manim/mobject/geometry.py

```diff
@@ -812,3 +859,233 @@ class RoundedRectangle(Rectangle):
+        # TODO: Once the config overhaul is far enough:
+        # Implement a way to apply kwargs to all tiles.
+        # The reason for this is that if multiple tilings
+        # are instantiated from one prototype, having a different basic
+        # tile setup is rather complicated now (set_fill etc).
+
+    def apply_transforms(self, x, y, tile):
+        """
+        Calls transform_tile once per dimension to position tiles.
+        Written like this to allow easy extending by Honeycomb.
+        """
+        self.transform_tile(x, self.x_offset, tile)
+        self.transform_tile(y, self.y_offset, tile)
+
+    def transform_tile(self, position, offset, tile):
+        """
+        This method computes and applies the offsets for the tiles,
+        in the given dimension.
+        multiplies inputs, which requires arrays to be numpy arrays.
+        """
+        # The number of different offsets the current axis has
+        offsets_nr = len(offset)
+        for i in range(offsets_nr):
+            for j in range(int(len(offset[i]) / 2)):
+                if position < 0:
+                    # Magnitude is calculated as the length of a range.
+                    # The range starts at 0, adjusting for the number
+                    # of different offset functions, stops at the target
+                    # position, and uses the amount of different
+                    # offset functions as the step.
+                    magnitude = len(range(-i, position, -offsets_nr)) * -1
+                    offset[-1 - i][0 + j * 2](
+                        tile, magnitude * np.array(offset[-1 - i][1 + j * 2])
+                    )
+                else:
+                    magnitude = len(range(i, position, offsets_nr))
+                    offset[i][0 + j * 2](
+                        tile, magnitude * np.array(offset[i][1 + j * 2])
+                    )
+
+
+class EdgeVertexGraph:
+    """
+    This class is for visual representation of graphs for graph theory.
+    (Not graphs of functions. Same term but entirely different things.)
+
+    It's instantiated with a dictionary that represents the graph, and
+    optionally which types of Mobject to use as vertices/edges and
+    dicts for their standard attributes.
+    
+    Parameters
+    ----------
+    graph : dict
+    vertex_type : MobjectClass, optional (default: Circle)
+    vertex_config : dict, optional
+    edge_type : MobjectClass, optional (default: ArcBetweenPoints)
+    edge_config : dict, optional
+
+    
+    The keys for the graph have to be of type int in ascending order,
+    with each number denoting a vertex.
+    The values are lists with 3 elements. A list of coordinates,
+    a list of connected vertices, and a configuration dictionary.
+    The coordinates determine the position of the vertex.
+    The list of connected vertices determines between which vertices
+    edges are. For clarity it's possible to have an edge defined in both
```

Comment:
This should specify that the list determines which neighbor vertices to which the vertex is connected. And from the code it seems like this can also be an int, so you should mention that here.

Reply:
It's enforced to be an int actually. I adapted the explanation above a bit.